api_app/tests_ma/test_api/test_routes/test_airlock.py

```diff
@@ -0,0 +1,252 @@
+import pytest
+from mock import patch
+from azure.eventgrid import EventGridEvent
+from fastapi import status
+from models.domain.airlock_review import AirlockReview, AirlockReviewDecision
+from db.errors import EntityDoesNotExist, UnableToAccessDatabase
+from models.domain.workspace import Workspace
+from models.domain.airlock_request import AirlockRequest, AirlockRequestStatus
+from tests_ma.test_api.test_routes.test_resource_helpers import FAKE_CREATE_TIMESTAMP
+from tests_ma.test_api.conftest import create_admin_user
+from azure.core.exceptions import HttpResponseError
+from resources import strings
+from services.authentication import get_current_workspace_owner_or_researcher_user_or_tre_admin, get_current_workspace_owner_or_researcher_user, get_current_workspace_owner_user
+pytestmark = pytest.mark.asyncio
+WORKSPACE_ID = "abc000d3-82da-4bfc-b6e9-9a7853ef753e"
+AIRLOCK_REQUEST_ID = "af89dccd-cdf8-4e47-8cfe-995faeac0f09"
+AIRLOCK_REVIEW_ID = "11bd2526-054b-4305-a7f9-63a2d6d2a80c"
+
+
+@pytest.fixture
+def sample_airlock_request_input_data():
+    return {
+        "requestType": "import",
+        "businessJustification": "some business justification"
+    }
+
+
+@pytest.fixture
+def sample_airlock_review_input_data():
+    return {
+        "reviewDecision": "approved",
+        "decisionExplanation": "the reason why this request was approved/rejected"
+    }
+
+
+def sample_status_changed_event(airlock_request_id=AIRLOCK_REQUEST_ID, workspace_id=WORKSPACE_ID):
+    eventgrid_event = EventGridEvent(
+        event_type="statusChanged",
+        data={
+            "request_id": airlock_request_id,
+            "status": "draft",
+            "type": "import",
+            "workspace_id": workspace_id[:-4]
+        },
+        subject=f"{airlock_request_id}/statusChanged",
+        data_version="2.0"
+    )
+    return eventgrid_event
+
+
+def sample_airlock_request_object(airlock_request_id=AIRLOCK_REQUEST_ID, workspace_id=WORKSPACE_ID):
+    airlock_request = AirlockRequest(
+        id=airlock_request_id,
+        workspaceId=workspace_id,
+        businessJustification="test business justification",
+        requestType="import"
+    )
+    return airlock_request
+
+
+def sample_submitted_airlock_request_object(airlock_request_id=AIRLOCK_REQUEST_ID, workspace_id=WORKSPACE_ID):
+    updated_airlock_request = AirlockRequest(
+        id=airlock_request_id,
+        workspaceId=workspace_id,
+        businessJustification="test business justification",
+        requestType="import",
+        status=AirlockRequestStatus.Submitted
+    )
+    return updated_airlock_request
+
+
+def sample_approved_airlock_request_object(airlock_request_id=AIRLOCK_REQUEST_ID, workspace_id=WORKSPACE_ID):
```

Comment:
if they just differ by status, make it an argument for the tests to pass

Reply:
Done